Nasm.md

```diff
@@ -252,3 +252,68 @@ je .value3_case
 ```
 
 ## Data structures
+
+As you can imagine assembly doesn't have intrinsic concept of data structures, but luckily we have the support of the preprocessor where that let us declare struct like blocks.
+
+This guide will just introduce quickly how to define a basic struct,for more information and use cases is better to check the netwide assembler official documentation (see the useful links section)
+
+Let's for example assume we have the following C struct:
+
+```c
+struct task {
+    uint32_t id;
+    char name[8];
+};
+```
+
+How nasm render a struct is basically declaring a list of offset labels in this way  we can use them to access the field starting from the struct memory location (*Authors note: yeah it is a trick...*)
+To create a struct in nasm we use the `struc` and `endstruc` keywords, and the fields are defined between these them. 
+The example above can be rendered in the following way:
+
+```asm
+struc task
+    id:         resd    1
+    name:       resd 8
```

Comment:
This shold be `resb` I think :P

Reply:
yeah lol i originally wrote resb, than changed the strucuture few times to end back to returning to the first version, but forgot to change it back XD